lib/src/clojure_lsp/kondo.clj

```diff
@@ -162,30 +157,38 @@
 
 (defn ^:private normalize
   "Put kondo result in a standard format, with `analysis` normalized and
-  `analysis` and `findings` indexed by filename."
+  `analysis` and `findings` indexed by uri."
   [{:keys [analysis findings] :as kondo-results}
-   {:keys [external? ensure-filenames filter-analysis] :or {filter-analysis identity}}]
-  (let [analysis (->> analysis
+   {:keys [external? ensure-uris filter-analysis] :or {filter-analysis identity}}
+   db]
+  (let [filename->uri (memoize #(shared/filename->uri % db))
```

Comment:
This conversion is a new, and potentially expensive, step in normalization. I benchmarked startup and ongoing analysis in some large projects and it didn't seem to make an overall difference, partially because we used to incur this same expense in the dep-graph code, and no longer do. Still, memoization seems prudent, to avoid re-calculating the same URI over and over.

Reply:
Looks ok, but when we memoize we memoize by all the args of the function right? so we would memorize given that db which could be a huge map, and memoizing lots of URIs for the same big db could affect application memory, right?

Reply:
But I suppose you tested performance because of that?

Reply:
Hum, this function is called once per call, so memoizing once may not be that bad I guess

Reply:
Yeah, this is a little tricky to read, but it seems to keep performance good and doesn't affect memory.
1. You're correct, we create a memoized function every time kondo is called. After we're done normalizing the analysis data from that one run we discard the memoized function, releasing the memory consumed by its arg/return-value cache. On the next run, we create a new memoized function. Using `memoize` seemed easier to read than using `loop` or `reduce` to collect a hashmap of filenames to uris. We've used that trick elsewhere to reduce the number of conversions, and it would be possible here too, just harder to read. The other benefit of `memoize` is that its cache is shared between the analysis normalization and the findings normalization. Again, the `loop` or `reduce` trick could do the same thing, it'd just be trickier.
2. The thing that is getting memoized is a function of one argument—the filename. The `db` is closed over, but isn't an argument, so the db isn't part of the cache key. But even if it were, as noted above, that cache is discarded at the end of normalization.

Reply:
Oh, that makes sense, so it's completely fine!